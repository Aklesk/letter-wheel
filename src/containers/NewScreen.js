import React from 'react';
import PropTypes from 'prop-types';
import {
  Alert,
  Text,
  View,
} from 'react-native';
import { inject, observer } from 'mobx-react';
import { SegmentedControls } from 'react-native-radio-buttons';
import CheckBox from 'react-native-checkbox';
import Button from 'react-native-button';

import WrapperStyle from '../styles/WrapperStyle';
import NewScreenStyle from '../styles/NewScreenStyle';

const options = [
  {
    label: '1-30\nWords',
    min: 1,
    max: 30,
    value: 0,
  },
  {
    label: '31-75\nWords',
    min: 31,
    max: 75,
    value: 1,
  },
  {
    label: '76+\nWords',
    min: 76,
    max: 999,
    value: 2,
  },
];

@inject('appStore') @observer
export default class LoadingScreen extends React.Component {
  static propTypes = {
    appStore: PropTypes.shape({
      nav: PropTypes.shape({
        goto: PropTypes.func.isRequired,
      }).isRequired,
    }).isRequired,
  }
  constructor(props) {
    super(props);
    this.state = {
      wordSelection: 0,
      timed: false,
    };
  }
  setSelectedOption = (wordSelection) => {
    this.setState({ wordSelection: wordSelection.value });
  }
  setTimed = (timed) => {
    this.setState({ timed: !timed });
  }
  instructions() { // eslint-disable-line class-methods-use-this
    Alert.alert('Coming soon', 'Instructions are still under development and will be implemented soon.');
  }
  start = () => {
    const { appStore } = this.props;
    const start = new Date().getTime();
    appStore.loading = true;
    appStore.newGame({
      wordsMin: options[this.state.wordSelection].min,
      wordsMax: options[this.state.wordSelection].max,
      timer: this.state.timed ? 60 : -1,
    })
      .then(() => {
        // Wait a minimum of 500ms to help the loading screen feel right
        const end = new Date().getTime();
        setTimeout(() => {
          appStore.nav.goto('Game');
          appStore.loading = false;
        }, (500 - (end - start)));
      });
  }
  render() {
    const {
      start,
      instructions,
      setSelectedOption,
      setTimed,
    } = this;
    const { timed } = this.state;
    const tint = '#555';
    return (
      <View style={WrapperStyle.container}>
        <View style={NewScreenStyle.header}>
          <Text style={NewScreenStyle.headerText}>
            NEW GAME
          </Text>
        </View>
        <View style={NewScreenStyle.body}>
          <View style={NewScreenStyle.words}>
            <SegmentedControls
              containerBorderTint={tint}
              containerStyle={NewScreenStyle.wordContainer}
              separatorTint={tint}
              tint={tint}
              options={options}
              onSelection={setSelectedOption}
              selectedIndex={this.state.wordSelection}
              optionStyle={NewScreenStyle.wordItems}
              extractText={option => option.label}
            />
          </View>
          <View style={NewScreenStyle.timer}>
            <CheckBox
              label={'Timed Game:'}
              labelBefore
              labelStyle={NewScreenStyle.timerLabel}
              checked={timed}
              onChange={setTimed}
            />
          </View>
          <View style={NewScreenStyle.buttonWrapper}>
            <Button
              style={NewScreenStyle.buttons}
              onPress={instructions}
            >
              Instructions
            </Button>
            <Button
              style={NewScreenStyle.buttons}
              onPress={start}
            >
              Start Game
            </Button>
          </View>
        </View>
      </View>
    );
  }
}
