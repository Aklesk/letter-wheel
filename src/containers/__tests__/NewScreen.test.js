import React from 'react';
import { mount } from 'enzyme';
import { runInAction } from 'mobx';

import AppStore from '../../data/store';
import NewScreen from '../NewScreen';

describe('New Screen component', () => {
  let store;

  beforeEach(() => {
    store = new AppStore();
  });

  afterEach(() => {
    jest.clearAllMocks();
    store = undefined;
  });

  it('matches the snapshot', () => {
    const render = mount(<NewScreen store={store} />);
    expect(render).toMatchSnapshot();
  });

  it('provides a callback for slider movement', () => {
    const render = mount(<NewScreen.wrappedComponent store={store} />);
    jest.spyOn(render.instance(), 'setSelectedOption');

    expect(render.instance().onSliderMove).toEqual(expect.any(Function));
    expect(render.instance().onSliderMove()).toEqual(undefined);
    expect(render.instance().setSelectedOption).toHaveBeenCalledTimes(0);

    render.instance().onSliderMove('test');
    expect(render.instance().setSelectedOption).toHaveBeenCalledTimes(0);

    render.instance().onSliderMove([1244, 6519]);
    expect(render.instance().setSelectedOption).toHaveBeenCalledTimes(1);
    expect(render.instance().setSelectedOption).toHaveBeenCalledWith({
      min: 1244,
      max: 6519,
    });

    jest.clearAllMocks();
    render.instance().onSliderMove([9152, 7627]);
    expect(render.instance().setSelectedOption).toHaveBeenCalledTimes(1);
    expect(render.instance().setSelectedOption).toHaveBeenCalledWith({
      min: 9152,
      max: 7627,
    });
  });

  it('provides a function to set range', () => {
    jest.spyOn(store, 'setNewGameOptions');

    const render = mount(<NewScreen.wrappedComponent store={store} />);

    expect(render.instance().setSelectedOption).toEqual(expect.any(Function));
    expect(render.instance().setSelectedOption({ min: 1, max: 645 })).toEqual(undefined);
    expect(store.setNewGameOptions).toHaveBeenCalledTimes(1);
    expect(store.setNewGameOptions).toHaveBeenCalledWith(
      'wordRange',
      {
        min: 1,
        max: 645,
      },
    );

    jest.clearAllMocks();
    render.instance().setSelectedOption({ min: 521, max: 1469 });
    expect(store.setNewGameOptions).toHaveBeenCalledTimes(1);
    expect(store.setNewGameOptions).toHaveBeenCalledWith(
      'wordRange',
      {
        min: 521,
        max: 1469,
      },
    );
  });

  it('provides a function to find whether the range matches a preset', () => {
    const render = mount(<NewScreen.wrappedComponent store={store} />);

    expect(render.instance().getSelectedOption).toEqual(expect.any(Function));
    expect(render.instance().getSelectedOption()).toEqual(0);

    runInAction(() => { store.setNewGameOptions('wordRange', { min: 0, max: 10 }); });
    expect(render.instance().getSelectedOption()).toEqual(-1);

    runInAction(() => { store.setNewGameOptions('wordRange', { min: 10, max: 49 }); });
    expect(render.instance().getSelectedOption()).toEqual(0);

    runInAction(() => { store.setNewGameOptions('wordRange', { min: 11, max: 49 }); });
    expect(render.instance().getSelectedOption()).toEqual(-1);

    runInAction(() => { store.setNewGameOptions('wordRange', { min: 50, max: 99 }); });
    expect(render.instance().getSelectedOption()).toEqual(1);

    runInAction(() => { store.setNewGameOptions('wordRange', { min: 50, max: 98 }); });
    expect(render.instance().getSelectedOption()).toEqual(-1);

    runInAction(() => { store.setNewGameOptions('wordRange', { min: 100, max: 999 }); });
    expect(render.instance().getSelectedOption()).toEqual(2);

    runInAction(() => { store.setNewGameOptions('wordRange', { min: 99, max: 999 }); });
    expect(render.instance().getSelectedOption()).toEqual(-1);
  });

  it('provides a function to get current screen width', () => {
    const render = mount(<NewScreen.wrappedComponent store={store} />);

    expect(render.instance().getSliderWidth).toEqual(expect.any(Function));
    expect(render.instance().getSliderWidth()).toEqual(330);

    runInAction(() => { store.width = 400; });
    expect(render.instance().getSliderWidth()).toEqual(330);
    runInAction(() => { store.width = 399; });
    expect(render.instance().getSliderWidth()).toEqual(329);
    runInAction(() => { store.width = 3333; });
    expect(render.instance().getSliderWidth()).toEqual(330);
    runInAction(() => { store.width = 250; });
    expect(render.instance().getSliderWidth()).toEqual(180);
  });

  it('provides a function to toggle timed state', () => {
    jest.spyOn(store, 'setNewGameOptions');

    const render = mount(<NewScreen.wrappedComponent store={store} />);

    expect(store.newGameOptions.get('timed')).toBeFalsy();
    expect(store.setNewGameOptions).toHaveBeenCalledTimes(0);
    render.instance().toggleTimed();
    expect(store.setNewGameOptions).toHaveBeenCalledTimes(1);
    expect(store.setNewGameOptions).toHaveBeenCalledWith('timed', true);
    expect(store.newGameOptions.get('timed')).toBeTruthy();
    render.instance().toggleTimed();
    expect(store.setNewGameOptions).toHaveBeenCalledTimes(2);
    expect(store.setNewGameOptions).toHaveBeenCalledWith('timed', false);
    expect(store.newGameOptions.get('timed')).toBeFalsy();
  });
});
