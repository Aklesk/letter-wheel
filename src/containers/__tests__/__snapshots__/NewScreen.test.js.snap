// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Screen component matches the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e8e9ed",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    style={
      Object {
        "width": 750,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignContent": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontSize": 50,
                "textAlign": "center",
              }
            }
          >
            NEW GAME
          </Text>
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "maxWidth": 400,
              "paddingHorizontal": 25,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "marginHorizontal": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderColor": "#555",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Array [
                      Object {
                        "backgroundColor": "#555",
                        "opacity": 1,
                        "paddingBottom": 5,
                        "paddingTop": 5,
                      },
                      Object {
                        "flex": 1,
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "fontSize": 20,
                        "paddingVertical": 10,
                      },
                      Object {
                        "color": "#ffffff",
                      },
                    ]
                  }
                >
                  10-49
                </Text>
              </View>
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Array [
                      Array [
                        Object {
                          "backgroundColor": "#ffffff",
                          "opacity": 1,
                          "paddingBottom": 5,
                          "paddingTop": 5,
                        },
                        Object {
                          "flex": 1,
                        },
                      ],
                      Object {
                        "borderLeftColor": "#555",
                        "borderLeftWidth": 1,
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "fontSize": 20,
                        "paddingVertical": 10,
                      },
                      Object {
                        "color": "#555",
                      },
                    ]
                  }
                >
                  50-99
                </Text>
              </View>
              <View
                accessible={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Array [
                      Array [
                        Object {
                          "backgroundColor": "#ffffff",
                          "opacity": 1,
                          "paddingBottom": 5,
                          "paddingTop": 5,
                        },
                        Object {
                          "flex": 1,
                        },
                      ],
                      Object {
                        "borderLeftColor": "#555",
                        "borderLeftWidth": 1,
                      },
                    ],
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "fontSize": 20,
                        "paddingVertical": 10,
                      },
                      Object {
                        "color": "#555",
                      },
                    ]
                  }
                >
                  100+
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "maxWidth": 400,
              "paddingHorizontal": 25,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
              }
            }
          >
            10 to 49 words
          </Text>
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "maxWidth": 400,
              "paddingHorizontal": 25,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 50,
                  "position": "relative",
                },
                Object {
                  "height": 30,
                  "overflow": "visible",
                  "paddingVertical": 20,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  Object {
                    "width": 330,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#A7A7A7",
                      "borderRadius": 2,
                      "height": 2,
                    },
                    Object {
                      "alignSelf": "center",
                      "borderRadius": 10,
                      "height": 10,
                    },
                    Object {
                      "backgroundColor": "#999",
                    },
                    Object {
                      "width": 0,
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#A7A7A7",
                      "borderRadius": 2,
                      "height": 2,
                    },
                    Object {
                      "alignSelf": "center",
                      "borderRadius": 10,
                      "height": 10,
                    },
                    Object {
                      "backgroundColor": "#444",
                    },
                    Object {
                      "width": 141.42857142857142,
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#A7A7A7",
                      "borderRadius": 2,
                      "height": 2,
                    },
                    Object {
                      "alignSelf": "center",
                      "borderRadius": 10,
                      "height": 10,
                    },
                    Object {
                      "backgroundColor": "#999",
                    },
                    Object {
                      "width": 188.57142857142858,
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "height": 48,
                      "justifyContent": "center",
                      "position": "absolute",
                      "width": 48,
                    },
                    Object {
                      "left": -24,
                      "top": -24,
                    },
                    false,
                  ]
                }
              >
                <View
                  onMoveShouldSetResponder={[Function]}
                  onMoveShouldSetResponderCapture={[Function]}
                  onResponderEnd={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderReject={[Function]}
                  onResponderRelease={[Function]}
                  onResponderStart={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  onStartShouldSetResponderCapture={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "stretch",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      Object {
                        "borderRadius": 15,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderColor": "#DDDDDD",
                            "borderRadius": 30,
                            "borderWidth": 1,
                            "height": 30,
                            "shadowColor": "#000000",
                            "shadowOffset": Object {
                              "height": 3,
                              "width": 0,
                            },
                            "shadowOpacity": 0.2,
                            "shadowRadius": 1,
                            "width": 30,
                          },
                          Array [
                            undefined,
                            Object {
                              "backgroundColor": "#fff",
                              "borderColor": "#555",
                              "borderRadius": 15,
                              "borderWidth": 2,
                              "height": 30,
                              "width": 30,
                            },
                          ],
                          undefined,
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "height": 48,
                      "justifyContent": "center",
                      "position": "absolute",
                      "width": 48,
                    },
                    Object {
                      "right": 164.57142857142858,
                      "top": -24,
                    },
                  ]
                }
              >
                <View
                  onMoveShouldSetResponder={[Function]}
                  onMoveShouldSetResponderCapture={[Function]}
                  onResponderEnd={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderReject={[Function]}
                  onResponderRelease={[Function]}
                  onResponderStart={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  onStartShouldSetResponderCapture={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "stretch",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      Object {
                        "borderRadius": 15,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "borderColor": "#DDDDDD",
                            "borderRadius": 30,
                            "borderWidth": 1,
                            "height": 30,
                            "shadowColor": "#000000",
                            "shadowOffset": Object {
                              "height": 3,
                              "width": 0,
                            },
                            "shadowOpacity": 0.2,
                            "shadowRadius": 1,
                            "width": 30,
                          },
                          Object {
                            "backgroundColor": "#fff",
                            "borderColor": "#555",
                            "borderRadius": 15,
                            "borderWidth": 2,
                            "height": 30,
                            "width": 30,
                          },
                          undefined,
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "maxWidth": 400,
              "paddingHorizontal": 25,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "marginHorizontal": 1,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "marginBottom": 5,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "marginLeft": 10,
                      "marginRight": 10,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "grey",
                          "fontSize": 15,
                        },
                        Object {
                          "color": "#000",
                          "fontSize": 20,
                        },
                      ]
                    }
                  >
                    Timed Game:
                  </Text>
                </View>
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "height": 26,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "maxWidth": 400,
              "paddingHorizontal": 25,
              "paddingVertical": 10,
            }
          }
        >
          <View
            accessibilityComponentType="button"
            accessibilityTraits="button"
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignContent": "center",
                "backgroundColor": "#999",
                "borderRadius": 4,
                "height": 60,
                "justifyContent": "center",
                "opacity": 1,
                "paddingHorizontal": 20,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007aff",
                    "fontSize": 17,
                    "fontWeight": "500",
                    "textAlign": "center",
                  },
                  null,
                  Object {
                    "color": "#000",
                    "fontSize": 20,
                    "lineHeight": 20,
                  },
                  null,
                ]
              }
            >
              Instructions
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />
          <View
            accessibilityComponentType="button"
            accessibilityTraits="button"
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignContent": "center",
                "backgroundColor": "#999",
                "borderRadius": 4,
                "height": 60,
                "justifyContent": "center",
                "opacity": 1,
                "paddingHorizontal": 20,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007aff",
                    "fontSize": 17,
                    "fontWeight": "500",
                    "textAlign": "center",
                  },
                  null,
                  Object {
                    "color": "#000",
                    "fontSize": 20,
                    "lineHeight": 20,
                  },
                  null,
                ]
              }
            >
              Start Game
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
